src/core/net/nat64_translator.hpp

```diff
@@ -163,7 +163,87 @@ class Translator : public InstanceLocator, private NonCopyable
      */
     void SetNat64Prefix(const Ip6::Prefix &aNat64Prefix);
 
+    /**
+     * @brief This function gets the next AddressMapping info (using an iterator).
+     *
+     * @param[in,out]  aIterator      the iterator. On success the iterator will be updated to point to next NAT64
+     * address mapping record. To get the first entry the iterator should be set to
+     * OT_NAT64_ADDRESS_MAPPING_ITERATOR_INIT.
+     * @param[out]     aMapping       an `otNat64AddressMapping` where information of next NAT64 address mapping record
+     * is placed (on success).
+     *
+     * @retval kErrorNone      Successfully found the next NAT64 address mapping info (@p aMapping was successfully
+     * updated).
+     * @retval kErrorNotFound  No subsequent NAT64 address mapping info was found.
+     *
+     */
+    Error GetNextAddressMapping(otNat64AddressMappingIterator &aIterator, otNat64AddressMapping &aMapping);
+
+    /**
+     * @brief This function gets the counters of the NAT64 translator, the counter is counted since the instance
+     * initialized.
+     *
+     * @param[out] aCounters An `otNat64Counters` where the counters of NAT64 translator will be placed.
+     *
+     */
+    void GetCounters(otNat64ProtocolCounters &aCounters) const { aCounters = mCounters; }
+
+    /**
+     * @brief This function gets the counters of the NAT64 translator, the counter is counted since the instance
+     * initialized.
+     *
+     * @param[out] aCounters An `otNat64Counters` where the counters of NAT64 translator will be placed.
+     *
+     */
+    void GetErrorCounters(otNat64ErrorCounters &aCounters) const { aCounters = mErrorCounters; }
+
+    /**
+     * Gets the configured CIDR in the NAT64 translator.
+     *
+     * @param[in]  aInstance    A pointer to an OpenThread instance.
+     * @param[out] aCidr        The `Ip4::Cidr` Where the configured CIDR will be placed.
+     *
+     * @retval kErrorNone       @p aCidr is set to the configured CIDR.
+     * @retval kErrorNotFound   The translator is not configured with an IPv4 CIDR.
+     *
+     */
+    Error GetIp4Cidr(Ip4::Cidr &aCidr);
+
+    /**
+     * Gets the configured CIDR in the NAT64 translator.
+     *
+     * @param[in]  aInstance         A pointer to an OpenThread instance.
+     * @param[out] aPrefix           The `Ip6::Prefix` where the configured NAT64 prefix will be placed.
+     *
+     * @retval kErrorNone       @p aPrefix is set to the configured prefix.
+     * @retval kErrorNotFound   The translator is not configured with an IPv6 prefix.
+     *
+     */
+    Error GetIp6Prefix(Ip6::Prefix &aPrefix);
+
 private:
+    class ProtocolCounters : public otNat64ProtocolCounters, public Clearable<ProtocolCounters>
```

Comment:
Move these to be `public`?
We cna then use then in methods instead of using teh C definitions `otNat64ProtocolCounters`
Can map the types using `DefineCoreType()`.